pumpkin-lib/src/branching/branchers/alternating_brancher.rs

```diff
@@ -143,8 +155,35 @@ impl<OtherBrancher: Brancher> Brancher for AlternatingBrancher<OtherBrancher> {
 
     fn on_restart(&mut self) {
         if let AlternatingStrategy::EveryRestart = self.strategy {
```

Comment:
It would be nice to be consistent in how we check for equality. Here, `if let ...` is used, but before `matches!` is used.